fvtest/compilertriltest/LinkageTest.cpp

```diff
@@ -151,3 +150,291 @@ TYPED_TEST(LinkageTest, SystemLinkageParameterPassingFourArg) {
+    if (compiler.compile() != 0) {
+        return NULL;
+    }
+
+    return compiler.getEntryPoint<T (*)(T,T,T,T)>();
+}
+
+TYPED_TEST(LinkageTest, SystemLinkageJitedToJitedParameterPassingFourArg) {
+    char inputTrees[400] = {0};
+
+    auto callee_entry_point = get_fourth_arg_from_callee<TypeParam>();
+    ASSERT_NOTNULL(callee_entry_point) << "Compilation of the callee failed unexpectedly\n";
+
+    EXPECT_EQ(static_cast<TypeParam>(1024),    callee_entry_point(0,0,0,static_cast<TypeParam>(1024)));
+    EXPECT_EQ(static_cast<TypeParam>(-1),      callee_entry_point(0,0,0,static_cast<TypeParam>(-1)));
+    EXPECT_EQ(static_cast<TypeParam>(0xf0f0f), callee_entry_point(0,0,0,static_cast<TypeParam>(0xf0f0f)));
+
+    const auto format_string = "(method return=%s args=[%s,%s,%s,%s] (block (%sreturn (%scall address=0x%jX args=[%s,%s,%s,%s] linkage=system"
+                                 " (%sload parm=0)"
+                                 " (%sload parm=1)"
+                                 " (%sload parm=2)"
+                                 " (%sload parm=3)"
+                                 ") )  ))";
+    std::snprintf(inputTrees, 400, format_string, TypeToString<TypeParam>::type,   // Return
+                                                  TypeToString<TypeParam>::type,   // Args
+                                                  TypeToString<TypeParam>::type,   // Args
+                                                  TypeToString<TypeParam>::type,   // Args
+                                                  TypeToString<TypeParam>::type,   // Args
+                                                  TypeToString<TypeParam>::prefix, // return
+                                                  TypeToString<TypeParam>::prefix, // call
+                                                  callee_entry_point,              // address
+                                                  TypeToString<TypeParam>::type,   // args
+                                                  TypeToString<TypeParam>::type,   // args
+                                                  TypeToString<TypeParam>::type,   // args
+                                                  TypeToString<TypeParam>::type,   // args
+                                                  TypeToString<TypeParam>::prefix, // load
+                                                  TypeToString<TypeParam>::prefix, // load
+                                                  TypeToString<TypeParam>::prefix, // load
+                                                  TypeToString<TypeParam>::prefix  // load
+                                                  );
+
+    auto trees = parseString(inputTrees);
+
+    ASSERT_NOTNULL(trees) << "Trees failed to parse\n" << inputTrees;
+
+    // Execution of this test is disabled on non-X86 platforms, as we
+    // do not have trampoline support, and so this call may be out of
+    // range for some architectures.
+#ifdef TR_TARGET_X86
```

Comment:
I'm not sure there is much value in executing everything that came before this. Please skip this test by adding the following at the beginning of the test case body:

```c
SKIP_IF(OMRPORT_ARCH_HAMMER == omrsysinfo_get_CPU_architecture(), MissingImplementation)
   << "Test is skipped on non-x86-64 platforms because calls are not currently supported on other platforms (see issue #1645)";
```

Reply:
FYI... `SKIP_IF` was recently introduced in #3539.

Reply:
Maybe you mean `SKIP_IF(OMRPORT_ARCH_HAMMER != omrsysinfo_get_CPU_architecture(), MissingImplementation)`? Otherwise test fails on my Windows X64. I see in `port\win32\omrsysinfo.c` the following code:

```cpp
#elif defined( _AMD64_)
	return OMRPORT_ARCH_HAMMER;
```